lib/Widgets/MessageDialog.vala

```diff
@@ -0,0 +1,261 @@
+     * @param buttons the {@link Gtk.ButtonsType} value that decides what buttons to use, defaults to {@link Gtk.ButtonsType.CLOSE},
+     *        see {@link Granite.MessageDialog.buttons} on details and what values are accepted
+     */
+    public MessageDialog (string primary_text, string secondary_text, GLib.Icon image_icon, Gtk.ButtonsType buttons = Gtk.ButtonsType.CLOSE) {
+        Object (
+            resizable: false,
+            deletable: false,
+            skip_taskbar_hint: true,
+            transient_for: null,
+            primary_text: primary_text,
+            secondary_text: secondary_text,
+            image_icon: image_icon,
+            buttons: buttons
+        );
+    }
+
+    /**
+     * Constructs a new {@link Granite.MessageDialog} with an icon name as it's icon displayed in the image.
+     * This constructor is same as the main one but with a difference that 
+     * you can pass an icon name string instead of manually creating the {@link GLib.Icon}.
+     * 
+     * The {@link Granite.MessageDialog.image_icon} will store the created icon
+     * so you can retrieve it later with {@link GLib.Icon.to_string}.
+     * 
+     * See {@link Gtk.Dialog} for more details.
+     * 
+     * @param primary_text the title of the dialog
+     * @param secondary_text the body of the dialog
+     * @param image_icon_name the icon name to create the dialog image with
+     * @param buttons the {@link Gtk.ButtonsType} value that decides what buttons to use, defaults to {@link Gtk.ButtonsType.CLOSE},
+     *        see {@link Granite.MessageDialog.buttons} on details and what values are accepted
+     */
+    public MessageDialog.with_image_from_icon_name (string primary_text, string secondary_text, string image_icon_name = "dialog-information", Gtk.ButtonsType buttons = Gtk.ButtonsType.CLOSE) {
+        Object (
+            resizable: false,
+            deletable: false,
+            skip_taskbar_hint: true,
+            transient_for: null,
+            primary_text: primary_text,
+            secondary_text: secondary_text,
+            image_icon: new ThemedIcon (image_icon_name),
+            buttons: buttons
+        );
+    }
+
+    construct {
+        image = new Gtk.Image ();
+        image.valign = Gtk.Align.START;
+
+        primary_label = new Gtk.Label (null);
+        primary_label.get_style_context ().add_class ("primary");
+        primary_label.selectable = true;
+        primary_label.max_width_chars = 50;
+        primary_label.wrap = true;
+        primary_label.xalign = 0;
+
+        secondary_label = new Gtk.Label (null);
+        secondary_label.use_markup = true;
+        secondary_label.selectable = true;
+        secondary_label.max_width_chars = 50;
+        secondary_label.wrap = true;
+        secondary_label.xalign = 0;
+
+        custom_bin = new SingleWidgetBin ();
+        custom_bin.add.connect (() => secondary_label.margin_bottom = 24);
```

Comment:
You have a `row_spacing` of 6, so you need to subtract that and have this margin be `18`